src/kvstore/raftex/RaftPart.h

```diff
@@ -486,37 +491,44 @@ class RaftPart : public std::enable_shared_from_this<RaftPart> {
 
     PromiseSet<AppendLogResult> sendingPromise_;
 
-    Status status_;
-    Role role_;
 
-    // When the partition is the leader, the leader_ is same as addr_
-    HostAddr leader_;
 
-    // The current term id
-    //
-    // When the partition voted for someone, termId will be set to
-    // the term id proposed by that candidate
-    TermID term_{0};
     // During normal operation, proposedTerm_ is equal to term_,
     // when the partition becomes a candidate, proposedTerm_ will be
     // bumped up by 1 every time when sending out the AskForVote
     // Request
     TermID proposedTerm_{0};
 
+    //***************************************************
+    // variables readed on keep-alive path, to avoid raftLock, we make them thread-safe
+    //
+    // The current term id
+    //
+    // When the partition voted for someone, termId will be set to
+    // the term id proposed by that candidate
+    TermID term_{0};
+    // When the partition is the leader, the leader_ is same as addr_
+    HostAddr leader_;
+    Status status_;
+    Role role_;
     // The id and term of the last-sent log
     LogID lastLogId_{0};
     TermID lastLogTerm_{0};
-    // The id for the last globally committed log (from the leader)
-    LogID committedLogId_{0};
-
+    folly::Synchronized<std::vector<std::shared_ptr<Host>>> hosts_;
```

Comment:
👍
Maybe we need to change `hosts_.size()` to `hosts->size()` in MemberChangeTest.cpp

Reply:
Good point.